rdflib/plugins/sparql/sparql.py

```diff
@@ -264,11 +265,27 @@ def __init__(
         self.graph: Optional[Graph]
         self._dataset: Optional[ConjunctiveGraph]
         if isinstance(graph, ConjunctiveGraph):
-            self._dataset = graph
-            if rdflib.plugins.sparql.SPARQL_DEFAULT_GRAPH_UNION:
-                self.graph = self.dataset
+            if datasetClause:
+                self._dataset = ConjunctiveGraph()
```

Comment:
Should this be `Dataset()`?

Reply:
Yes, `ConjunctiveGraph` is deprecated in 7.x for removal in 8.x, so shift all references to `Dataset`

Reply:
I changed dataset init with `Dataset` instead of `ConjunctiveGraph`. I didn't change `if isinstance(graph, ConjunctiveGraph):` so we can still query `ConjunctiveGraph` for the moment. I guess this should be changed when `ConjunctiveGraph` is totally removed

Reply:
Thanks!